vlasiator.cpp

```diff
@@ -791,10 +793,10 @@ int simulate(int argn,char* args[]) {
    // Invalidate cached cell lists just to be sure (might not be needed)
    P::meshRepartitioned = true;
 
-   unsigned int wallTimeRestartCounter=1;
+   uint wallTimeRestartCounter=1;
+   uint recoverCounter=0;
 
-   int doNow[3] = {0}; // 0: writeRestartNow, 1: balanceLoadNow, 2: refineNow ; declared outside main loop
-   int writeRestartNow; // declared outside main loop
+   int doNow[4] = {0}; // 0: writeRestartNow, 1: writeRecoverNow, 2: balanceLoadNow, 3: refineNow ; declared outside main loop
```

Comment:
For code readability, I would actually suggest to have a struct
```c++
struct DoNow {
   bool writeRestart;
   bool writeRecover;
   bool balanceLoad;
   bool refine;
} doNow;
```
instead. Or at least make the array entries preprocessor macros. Or an enum.

Because the hardcoded array index numbers in the actual logic code further down are sure to confuse the uninitiated.